fineract-loan/src/main/java/org/apache/fineract/portfolio/interestpauses/service/InterestPauseReadPlatformServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+            String locale) {
+        return processInterestPause(() -> {
+            Long loanId = loanRepositoryWrapper.findIdByExternalId(loanExternalId);
+            if (loanId == null) {
+                throw new LoanNotFoundException("Loan not found with External ID: " + loanExternalId);
+            }
+            return loanRepositoryWrapper.findOneWithNotFoundDetection(loanId);
+        }, startDate, endDate, dateFormat, locale);
+    }
+
+    @Override
+    @Transactional
+    public CommandProcessingResult createInterestPauseByLoanId(Long loanId, LocalDate startDate, LocalDate endDate, String dateFormat,
+            String locale) {
+        return processInterestPause(() -> loanRepositoryWrapper.findOneWithNotFoundDetection(loanId), startDate, endDate, dateFormat,
+                locale);
+    }
+
+    private CommandProcessingResult processInterestPause(Supplier<Loan> loanSupplier, LocalDate startDate, LocalDate endDate,
+            String dateFormat, String locale) {
+
+        this.context.authenticatedUser();
+        final Loan loan = loanSupplier.get();
+
+        validateInterestPauseDates(loan, startDate, endDate, dateFormat, locale);
+
+        LoanTermVariations variation = new LoanTermVariations(LoanTermVariationType.INTEREST_PAUSE.getValue(), startDate, null, endDate,
+                false, loan);
+
+        LoanTermVariations savedVariation = loanTermVariationsRepository.saveAndFlush(variation);
+
+        return new CommandProcessingResultBuilder().withEntityId(savedVariation.getId()).build();
+    }
+
+    private void validateInterestPauseDates(Loan loan, LocalDate startDate, LocalDate endDate, String dateFormat, String locale) {
+
+        validateOrThrow(baseDataValidator -> {
+            baseDataValidator.reset().parameter("startDate").value(startDate).notBlank();
+            baseDataValidator.reset().parameter("endDate").value(endDate).notBlank();
+            baseDataValidator.reset().parameter("dateFormat").value(dateFormat).notBlank();
+            baseDataValidator.reset().parameter("locale").value(locale).notBlank();
+        });
+
+        if (startDate.isBefore(loan.getSubmittedOnDate())) {
+            throw new GeneralPlatformDomainRuleException("interest.pause.start.date.before.loan.start.date",
+                    String.format("Interest pause start date (%s) cannot be earlier than loan start date (%s).", startDate,
+                            loan.getSubmittedOnDate()),
+                    startDate, loan.getSubmittedOnDate());
+        }
+
+        if (endDate.isAfter(loan.getMaturityDate())) {
+            throw new GeneralPlatformDomainRuleException("interest.pause.end.date.after.loan.maturity.date", String
+                    .format("Interest pause end date (%s) cannot be later than loan maturity date (%s).", endDate, loan.getMaturityDate()),
+                    endDate, loan.getMaturityDate());
+        }
+
+        if (!endDate.isAfter(startDate)) {
```

Comment:
Start date and End date can be the same, which means for 1 day the interest is paused!

Reply:
Fixed